catchup/service.go

```diff
@@ -848,26 +852,30 @@ func createPeerSelector(net network.GossipNode, cfg config.Local, pipelineFetch
 			if cfg.NetAddress != "" { // Relay node
 				peerClasses = []peerClass{
 					{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersConnectedOut},
-					{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookRelays},
-					{initialRank: peerRankInitialThirdPriority, peerClass: network.PeersConnectedIn},
+					{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookArchivalNodes},
+					{initialRank: peerRankInitialThirdPriority, peerClass: network.PeersPhonebookRelays},
+					{initialRank: peerRankInitialFourthPriority, peerClass: network.PeersConnectedIn},
 				}
 			} else {
 				peerClasses = []peerClass{
-					{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersConnectedOut},
-					{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookRelays},
+					{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookArchivalNodes},
+					{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersConnectedOut},
```

Comment:
This is the only one where the archival nodes are above the connectedOut peers

Reply:
I guess the class for lines 825 is also like this. 
Is there any kind of reasoning for the ordering of these things?  
this function maybe could be expressed cleaner.

Reply:
I think this is valid concern, @algonautshant ?

Reply:
Unfortunately, adding a 4th class does not make things cleaner. The limited classes (was 3, now 4) was to control the complexity. This is not an engineering decision, it is a management decision.

Now, regarding the reasoning: 
The cases on line 825 and 862 are the cases where we have a non-relay node making a `pipelineFetch`. This would be the least urgent case. `pipelineFetch` is by the catchup service, when the node is behind, and wants to catchup. Also, it is expected to be outside the non-archival block retention range. 

So, this category needs a block which is likely not available in a non-archival nodes, and it is not super urgent (not as urgent as the agreement service asking for a block, or as important to the network as a relay would be). A typical case would be a node catching up to join the network.
So, for such requests, the traffic is diverted to `PeersPhonebookArchivalNodes` and `PeersPhonebookArchivers` before going to `PeersConnectedOut` to leave the gossip network for higher priority operations, and not long running operations like catchup.

I hope this convinces you of the reasoning.

Reply:
right, now I recall we've already talked about this, thanks.